### ISSUE TITLE
secondary text pdf attachments (feature)

### ISSUE THREAD

**Issue by ghost** [melmothx/amusewiki#309]
i wanted to ask if its possible to associate a secondary pdf to a text, in my use case it wd be for example a real scan of the book, so that an archive could present both the book as originally published as well as the excellent existing output/export options. seems fairly standard for archive style projects. afaict at the moment the pdf attachments only work to embed images directly in the text?

**Comment by mouseb** [comment]
ahhh, ok i see how this works! of course this was already possible.

u just have to 'attach' the attachment after adding it to an entry.

but also, given that we are dealing with long books, would it be possible to make it available on a page somewhere else than at the very bottom? i actually imagined it wd go in the bar w the title and all other output options. so it doesn't feel like an afterthought.

**Comment by melmothx** [comment]
mouseb <notifications@github.com> writes:

> ahhh, ok i see how this works! of course this was already possible.
>
> u just have to 'attach' the attachment after adding it to an entry.
>
> but also, given that we are dealing with long books, would it be
> possible to make it available on a page somewhere else than at the
> very bottom? i actually imagined it wd go in the bar w the title and
> all other output options. so it doesn't feel like an afterthought.

I think this is doable and straightforward


--
Marco

**Comment by melmothx** [comment]
@mouseb See https://staging.amusewiki.org/library/test for a demo. I think this should be OK?

**Comment by mouseb** [comment]
yeah this looks great! i wdnt necessarily ruin your nice info bar w a larger pdf thumbnail, but up to you obvs. i also think it needs a kind of label to, to make clear that it is the scanned original. ah, or i just title it 'scanned original' and thats what the mouse-over gives right. very nice.

i guess this gallery also solves my covers attachment query? but in that case there is still the question of whether the cover files can be made avail in the bookbuilder as attachments...

then there wd be the full package: orig scan, marked up doc, covers. allowing the user to re-create a full booklet based on the orig.

**Comment by melmothx** [comment]
@mouseb cover attachment is another issue.

I'm probably going to improve the look when I get the time.

For the mouse-over, I think we should leverage the attachment title (you know you can add that), and that was done on the staging site. I don't see much point in adding an hardcoded "original" when it's just an arbitrary file attached.

**Comment by melmothx** [comment]
Actually, I think this is a good as it can get, so that would be it for now.

**Comment by rojenzaman** [comment]
is there any way to disable standard amusewiki pdf, epub, tex, html formats when using scanned pdfs? because it is unnecessary when i use only scanned pdf.

**Comment by melmothx** [comment]
Rojen Zaman ***@***.***> writes:

> is there any way to disable standard amusewiki pdf, epub, tex, html formats when using scanned pdfs? because it is unnecessary when i use only scanned pdf.

Try

#blob yes

In the header.

-- 
Marco

**Comment by rojenzaman** [comment]
> Try
>
> #blob yes
>
> In the header.

Thank you! I didn't know that.

**Comment by rojenzaman** [comment]
```
#blob yes
```

But when I use it, the text information does not appear.

**Comment by melmothx** [comment]
@rojenzaman it looks like GitHub is spitting out some old comments which at the time weren't delivered. At any rate, try then:

```
#formats none
```

And see if it does the trick.

**Comment by rojenzaman** [comment]
It still creates epub, html, tex and zip files.

**Comment by melmothx** [comment]
@rojenzaman Please try again, but switching branch to `no-formats-309`

**Comment by rojenzaman** [comment]
I've tried `no-formats-309` branch, but:

 - File icons are hidden even if `#formats none` is not specified. (on posts with attachments)
 - Still creates epub, html, tex and zip files.

**Comment by melmothx** [comment]
First point is 

> I've tried `no-formats-309` branch, but:
> 
>     * File icons are hidden even if `#formats none` is not specified. (on posts with attachments)

I believe you still have `#blob yes` set.

>     * Still creates epub, html, tex and zip files.

Yes, but I don't see the problem with that.

>     * .muse format/icon is hidden

If it's basically empty, what's the point of having it there?

**Comment by rojenzaman** [comment]
> I believe you still have `#blob yes` set.

No, but I didn't have this problem on a clean installation.

> If it's basically empty, what's the point of having it there?

Maybe someone would still want to see the .muse source, but that's okay.

**Comment by rojenzaman** [comment]
> Yes, but I don't see the problem with that.

Here was my problem:

If there are over 1000 posts and each one has an attached file, why create epub, pdf, tex, zip files for them one by one? Being able to specify the file creation (like #createformats no) relieves the load and saves time.

**Comment by melmothx** [comment]
Rojen Zaman ***@***.***> writes:

> Here was my problem:
>
> If there are over 1000 posts and each one has an attached file, why
> create epub, pdf, tex, zip files for them one by one? Being able to
> specify the file creation (like #createformats no) relieves the load
> and saves time.

Well, no PDF is generated with #formats none, and the PDF generation is
the real one which waste time and space.

I'm pretty sure we're relying on the HTML for at least indexing, so
that's needed. EPUB is just html, but compressed. TeX alone is just
text. Zip: that's needed for the bookbuilder, which in this case may or
may not be needed.

My line of thought is that if you really have an instance with 1000
texts, all of them just basic info with some blobs attached, then yes,
the request is legit and it's worth looking at.

However, if this is just a remote possibility, then this is premature
optimization and complicates the flow, breaks assumptions (like
HTML/epub/zip always been there) just to save a few bytes on the disk
and a few seconds (if not less) when publishing a text.

-- 
Marco

**Comment by rojenzaman** [comment]
OK. It's not worth the development time, it would be a bad idea to complicate the flow. Maybe the issue can be closed, thank u.

**Comment by melmothx** [comment]
Calculated saving would be 20KB per text and sub-second saving at publish time.
